Spokestack/SpokestackSpeechRecognizer.swift

```diff
@@ -106,8 +106,13 @@ import CryptoKit
     private func receive(result: (Result<URLSessionWebSocketTask.Message, Error>)) {
         self.handle(result, handleResult: { r in
             if let hypothesis = r.hypotheses.first {
-                self.context.confidence = hypothesis.confidence
-                self.context.transcript = hypothesis.transcript
+                if !hypothesis.transcript.isEmpty {
+                    if hypothesis.transcript != self.context.transcript {
```

Comment:
An extra string equality check feels like unnecessary overhead to me—both the chances of the system producing a bunch of identical transcripts in a row, and the cost of receiving duplicate partial results seem low.

Reply:
> the chances of the system producing a bunch of identical transcripts in a row
this is exactly why I had to do it. otherwise it's just walls of either empty or duplicated partial transcripts

Reply:
Huh. Interesting. I don't remember many duplicates in my testing, but maybe I wasn't creative enough with my utterances.